demo/vite.config.ts

```diff
@@ -1,110 +1,111 @@
+              if (pathname.endsWith('.html')) {
+                res.setHeader('Content-Type', 'text/html')
+                res.writeHead(200)
+                res.write(fs.readFileSync(path.join(__dirname, pathname)))
+                res.end()
+              }
+            }
+
+            next()
+          })
+        }
+      }
+    }
+  ],
+  optimizeDeps: {
+    // This is require because vite excludes local dependencies from being optimized
+    // Monaco-vscode-api packages are local dependencies and the number of modules makes chrome hang
+    include: [
+      // add all local dependencies...
+      ...localDependencies,
+      // and their exports
+      'vscode/extensions', 'vscode/services', 'vscode/monaco', 'vscode/localExtensionHost',
+
+      // These 2 lines prevent vite from reloading the whole page when starting a worker (so 2 times in a row after cleaning the vite cache - for the editor then the textmate workers)
+      // it's mainly empirical and probably not the best way, fix me if you find a better way
+      'monaco-editor/esm/vs/nls.js', 'monaco-editor/esm/vs/editor/editor.worker.js', 'vscode-textmate', 'vscode-oniguruma', '@vscode/vscode-languagedetection', 'vscode-semver',
+      ...(await glob('monaco-editor/esm/vs/**/common/**/*.js', { cwd: path.resolve(__dirname, '../node_modules') }))
+    ],
+    exclude: [],
+    esbuildOptions: {
+      tsconfig: './tsconfig.json',
+      plugins: [{
+        name: 'import.meta.url',
+        setup ({ onLoad }) {
+          // Help vite that bundles/move files in dev mode without touching `import.meta.url` which breaks asset urls
+          onLoad({ filter: /.*\.js/, namespace: 'file' }, async args => {
+            const code = fs.readFileSync(args.path, 'utf8')
+
+            const assetImportMetaUrlRE = /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*,\s*import\.meta\.url\s*(?:,\s*)?\)/g
+            let i = 0
+            let newCode = ''
+            for (let match = assetImportMetaUrlRE.exec(code); match != null; match = assetImportMetaUrlRE.exec(code)) {
+              newCode += code.slice(i, match.index)
+
+              const path = match[1].slice(1, -1)
+              const resolved = await resolve(path, url.pathToFileURL(args.path).toString())
+
+              newCode += `new URL(${JSON.stringify(url.fileURLToPath(resolved))}, import.meta.url)`
+
+              i = assetImportMetaUrlRE.lastIndex
+            }
+            newCode += code.slice(i)
+
+            return { contents: newCode }
+          })
+        }
+      }]
+    }
+  },
+  server: {
+    port: 5173,
+    origin: cdnDomain,
+    host: '0.0.0.0',
+    fs: {
+      allow: ['../'] // allow to load codicon.ttf from monaco-editor in the parent folder
+    }
+  },
+  define: {
+    rootDirectory: JSON.stringify(__dirname)
+  },
+  resolve: {
+    dedupe: ['monaco-editor', 'vscode', ...localDependencies]
+  }
+})

```

Comment:
Why is github not able to display a proper diff on this file?

Reply:
I may have messed up the line endings. I can fix it now, give me a few minutes. I'll rebase while I'm at it.

Reply:
Did you forgot to push? :)

Reply:
It still doesn't seem to be ok :thinking: